lang/compiler_test.go

```diff
@@ -1126,3 +1126,75 @@ var crlfPattern = regexp.MustCompile(`\r?\n`)
 func toCRLF(data string) string {
 	return crlfPattern.ReplaceAllString(data, "\r\n")
 }
+
+func TestAstProgram_RecordedEvents(t *testing.T) {
+	now := time.Now()
+	for _, tt := range []struct {
+		name string
+		src  string
+		key  string
+	}{
+		{
+			name: "window_groupbytime",
+			src: `import "internal/gen"
+gen.tables(n: 30)
+|> range(start: -5m)
+|> group(columns: ["_time"])
+|> window(every: 1m, createEmpty: false)
+`,
+			key: "window/time-in-group-key",
+		},
+		{
+			name: "aggregatewindow_groupbytime",
+			src: `import "internal/gen"
+gen.tables(n: 30)
+|> range(start: -5m)
+|> group(columns: ["_time"])
+|> aggregateWindow(every: 1m, fn: mean, createEmpty: false)
+`,
+			key: "window/time-in-group-key",
+		},
+		{
+			name: "disjoint_tablefind",
+			src: `import "internal/gen"
+a = gen.tables(n: 10)
+b = a
+	|> tableFind(fn: (key) => true)
+	|> getRecord(idx: 0)
+a
+	|> map(fn: (r) => ({_time: r._time, _value: r._value + b._value}))
+	|> yield(name: "_result")
```

Comment:
I'm trying to reconcile why this Flux code would trigger the event. I can see that the call to `gen.tables()` would be executed more than once for this plan because it is consumed by `tableFind` and also yielded.

It seems to me that because `buildSpec` checks that parents are not yet visited before it recurses to them, doesn't that mean it would fail to find the conflict? But this test still passes, so I must be missing something.

I would expect that:
-

Reply:
`buildSpec` is invoked multiple times so the visited map only lives through each construction of a spec rather than the construction of all specs.

table find forces a spec to be built and another spec is built with yield. So `gen.tables()` is visited twice.